src/main/java/org/candlepin/model/ContentCurator.java

```diff
@@ -688,4 +699,84 @@ public Map<Content, Boolean> getActiveContentByOwner(String ownerId) {
 
         return activeContent;
     }
+
+    /**
+     * Fetches a collection of content based on the data in the query builder. If the
+     * query builder is null or contains no arguments, the query will not limit or sort the result.
+     *
+     * @param queryArgs
+     *     a ContentQueryArguments instance containing the various arguments to use to
+     *     select contents
+     *
+     * @return a list of contents. It will be paged and sorted if specified
+     */
+    public List<Content> listAll(ContentQueryArguments queryArgs) {
+        CriteriaBuilder criteriaBuilder = this.getEntityManager().getCriteriaBuilder();
+        CriteriaQuery<Content> criteriaQuery = criteriaBuilder.createQuery(Content.class);
+
+        Root<Content> root = criteriaQuery.from(Content.class);
+        criteriaQuery.select(root)
+            .distinct(true);
+
+        List<Predicate> predicates = new ArrayList<>();
+        Predicate securityPredicate = this.getSecurityPredicate(Content.class, criteriaBuilder, root);
+        if (securityPredicate != null) {
+            predicates.add(securityPredicate);
+        }
+
+        if (!predicates.isEmpty()) {
+            criteriaQuery.where(predicates.toArray(new Predicate[predicates.size()]));
+        }
```

Comment:
Since the predicates list will be one element at max we can just check if securityPredicate is not null and call `criteriaQuery.where()` if it is not null. This avoids creating a list and then moving the element from the list to an array.

```
Predicate securityPredicate = this.getSecurityPredicate(DeletedConsumer.class, criteriaBuilder,
    root);
if (securityPredicate != null) {
    criteriaQuery.where(securityPredicate);
}
```

This can also be applied in the `getContentCount` method.